compiler/src/main/java/com/squareup/anvil/compiler/codegen/Contribution.kt

```diff
@@ -197,6 +197,40 @@ internal sealed class Contribution {
       .thenComparing(compareBy { it.boundType.canonicalName })
       .thenComparing(compareBy { if (it is Binding) it.rank else 0 })
       .thenComparing(compareBy { it.replaces.joinToString(transform = ClassName::canonicalName) })
+
+    internal fun uniqueTypeName(
+      originType: ClassName,
+      boundType: ClassName,
+      scopeType: ClassName,
+      suffix: String,
+    ): String {
+
+      val md5Hash = MessageDigest.getInstance("MD5")
+        .apply {
+          update(originType.canonicalName.toByteArray())
+          update(boundType.canonicalName.toByteArray())
+          update(scopeType.canonicalName.toByteArray())
+          update(suffix.toByteArray())
+        }
+        .digest()
+        .take(8)
+        .joinToString("") { "%02X".format(it) }
```

Comment:
sidenote - okio has some nice APIs for this

Reply:
I've been tempted to add Okio, and also kotlinx-serialization or Moshi for the incremental caching.  But that would mean adding things to the compiler classpath and/or adding the shadow plugin to our build, and it just didn't seem to be worth the hassle.